### PR TITLE
Improve instance label validation and behaviour.

### PR DESCRIPTION
**Pull request by dracos**
Increase the maximum label length to 63, allow labels to start with a
number, but don't allow them to end with a hyphen. Explain that hyphens
are allowed in the validation text.

### CODE REVIEW

**Comment by chrismytton** [comment]
It appears that despite me saying that this *should* work in IRC I forgot to take into account the restriction that MongoDB places on the database name length:

> [Database names cannot be empty and must have fewer than 64 characters](http://docs.mongodb.org/manual/reference/limits/#Length-of-Database-Names)

So actually the instance slug needs to take into account the length of [`config.MongoDB.popit_prefix`](https://github.com/mysociety/popit/blob/543b4bb184ff4191bd76d96a4830b7905c800b0f/config/default.js#L46) for the current environment.

File: hosting-app/views/instance_new.html [hunk 1]
```diff
@@ -28,7 +28,10 @@
       <% if (errors.slug == 'slug_not_unique' ) { %>
       <span class="help-block">A Popit instance with that name already exists at <a href="<%= existing_instance_url %>"><%= existing_instance_url %></a></span>
       <% } else if (errors.slug) { %>
-        <span class="help-block">Your website name can only contain letters and numbers. It should be at least four letters long but no longer than 20.</span>
+        <span class="help-block">Your website name can only contain letters,
+        numbers, and hyphens. It should be at least four letters long but no
+        longer than <%= (63 - config.MongoDB.popit_prefix.length) ?>.</span>
```
Comment (chrismytton) [inline hosting-app/views/instance_new.html]: I think the travis tests are failing because the closing tag here needs to be `%>` rather than `?>`.

Comment (dracos) [inline hosting-app/views/instance_new.html]: Whoops!

**Comment by chrismytton** [comment]
Looks good! :+1:

**Pull request merged**